engines/router/missionctl/handlers/batch_http_handler.go

```diff
@@ -0,0 +1,106 @@
+package handlers
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"github.com/gojek/turing/engines/router/missionctl"
+	"github.com/gojek/turing/engines/router/missionctl/errors"
+	"github.com/gojek/turing/engines/router/missionctl/instrumentation/tracing"
+	"github.com/gojek/turing/engines/router/missionctl/log"
+	"github.com/gojek/turing/engines/router/missionctl/turingctx"
+)
+
+type batchHTTPHandler struct {
+	httpHandler
+}
+
+type batchResponse struct {
+	StatusCode int             `json:"code"`
+	ErrorMsg   string          `json:"error,omitempty"`
+	Data       json.RawMessage `json:"data,omitempty"`
+}
+
+// NewBatchHTTPHandler creates an instance of the Mission Control's prediction request handler
+func NewBatchHTTPHandler(mc missionctl.MissionControl) http.Handler {
+	return &batchHTTPHandler{httpHandler{mc}}
+}
+
+func (h *batchHTTPHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	var httpErr *errors.HTTPError
+	h.measureRequestDuration(httpErr)
+
+	// Create context from the request context
+	ctx := turingctx.NewTuringContext(req.Context())
+	// Create context logger
+	ctxLogger := log.WithContext(ctx)
+	defer func() {
+		_ = ctxLogger.Sync()
+	}()
+
+	// Get the unique turing request id from the context
+	turingReqID, err := turingctx.GetRequestID(ctx)
+	if err != nil {
+		ctxLogger.Errorf("Could not retrieve Turing Request ID from context: %v",
+			err.Error())
+	}
+	ctxLogger.Debugf("Received batch request for %v", turingReqID)
+
+	if tracing.Glob().IsEnabled() {
+		ctx = h.enableTracingSpan(ctx, req)
```

Comment:
Can you make sure that distributed tracing works as intended with the batch request to Turing? I.e. that the tracing headers are propagated to processing individual requests and separate nested tracing spans are created for each individual request?

Reply:
checked.
![Screenshot 2021-11-24 at 7 41 09 PM](https://user-images.githubusercontent.com/30390872/143338205-de8338ef-f29c-4608-8185-9f8cbe4e03d1.png)
![Screenshot 2021-11-24 at 7 13 21 PM](https://user-images.githubusercontent.com/30390872/143338365-53d15971-cab4-4c2f-a718-28c9ec7462ee.png)